tools/tests/test_fan.py

```
@@ -0,0 +1,196 @@
+    Timestamps were derived from `enumerate(demux(...))` while the clip length was
+    derived from the *kept* packet count. A skipped zero-size packet made the
+    length shorter than the real span, so _pts_offset advanced by less than one
+    clip and the second loop's timestamps overlapped the first.
+    """
+    fps = 30
+    step = int(90000 / fps)
+
+    # Replay the demux loop exactly as _ensure_clip does, with a zero-size packet
+    # in the middle -- which is what ffmpeg emits for a flush.
+    class FakePacket:
+        def __init__(self, size):
```

Ruff (0.16.1)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)